Source/CarthageKit/GitHub.swift

```diff
@@ -266,6 +284,14 @@ internal struct GitHubCredentials {
 	}
 }
 
+internal func loadGitHubCredentials(accessToken: GitHubAccessToken?) -> SignalProducer<GitHubCredentials?, CarthageError> {
```

Comment:
> Also we would have to change every method that receives as GitHubCredentials to receive a String with the value of the authorization header.

There's only a few methods in the project that takes a `GitHubCredentials`. They all pass them through to `createGitHubRequest`, which only uses them to get the `authorizationHeaderValue`.

If you think having the type is preferable, then how about we make `GitHubCredentials` an enum?

```swift
internal enum GitHubCredentials {
    case .Basic(username: String, password: String)
    case .Token(String)

    var authorizationHeaderValue: String { ... }
}
```